File: Modul-4/Python/PRAK404_2410817210001_MuhammadFadillah.py
def kalkulator():
    while True:
        try:
            print("\nPilih program:")
            print("1. Penjumlahan")
            print("2. Pengurangan")
            print("3. Perkalian")
            print("4. Pembagian")
            print("5. Exit")
            pilihan = int(input("Masukkan Pilihan: "))
            if pilihan in [1, 2, 3, 4]:
                nilai_pertama = float(input("Masukkan nilai pertama: "))
                nilai_kedua = float(input("Masukkan nilai kedua: "))

                if pilihan == 1:
                    hasil = nilai_pertama + nilai_kedua
                    print(f"Hasil Penjumlahan antara {nilai_pertama:.2f} dan {nilai_kedua:.2f} adalah {hasil:.2f}")
                elif pilihan == 2:
                    hasil = nilai_pertama - nilai_kedua
                    print(f"Hasil Pengurangan antara {nilai_pertama:.2f} dan {nilai_kedua:.2f} adalah {hasil:.2f}")
                elif pilihan == 3:
                    hasil = nilai_pertama * nilai_kedua
                    print(f"Hasil Perkalian antara {nilai_pertama:.2f} dan {nilai_kedua:.2f} adalah {hasil:.2f}")
                elif pilihan == 4:
                    if nilai_kedua != 0:
                        hasil = nilai_pertama / nilai_kedua
                        print(f"Hasil Pembagian antara {nilai_pertama:.2f} dan {nilai_kedua:.2f} adalah {hasil:.2f}")
                    else:
                        print("Error: Pembagian dengan nol tidak diperbolehkan.")
            elif pilihan == 5:
                print("Terimakasih, telah menggunakan kalkulator Muhammad Fadillah.")
                break
            else:
                print("Input anda salah, silahkan coba lagi.")
        except ValueError:
            print("Input tidak valid, masukkan angka yang sesuai.")

kalkulator()

File: Modul-4/C/PRAK404_2410817210001_MuhammadFadillah.c
#include <stdio.h>

int main() {
    int pilihan;
    float nilaiPertama, nilaiKedua, hasil;

    do {
        printf("\nPilih program:\n");
        printf("1. Penjumlahan\n");
        printf("2. Pengurangan\n");
        printf("3. Perkalian\n");
        printf("4. Pembagian\n");
        printf("5. Exit\n");
        printf("Masukkan Pilihan: ");
        scanf("%d", &pilihan);

        if (pilihan >= 1 && pilihan <= 4) {
            printf("Masukkan nilai pertama: ");
            scanf("%f", &nilaiPertama);
            printf("Masukkan nilai kedua: ");
            scanf("%f", &nilaiKedua);

            switch (pilihan) {
                case 1:
                    hasil = nilaiPertama + nilaiKedua;
                    printf("Hasil Penjumlahan antara %.2f dan %.2f adalah %.2f\n", nilaiPertama, nilaiKedua, hasil);
                    break;
                case 2:
                    hasil = nilaiPertama - nilaiKedua;
                    printf("Hasil Pengurangan antara %.2f dan %.2f adalah %.2f\n", nilaiPertama, nilaiKedua, hasil);
                    break;
                case 3:
                    hasil = nilaiPertama * nilaiKedua;
                    printf("Hasil Perkalian antara %.2f dan %.2f adalah %.2f\n", nilaiPertama, nilaiKedua, hasil);
                    break;
                case 4:
                    if (nilaiKedua != 0) {
                        hasil = nilaiPertama / nilaiKedua;
                        printf("Hasil Pembagian antara %.2f dan %.2f adalah %.2f\n", nilaiPertama, nilaiKedua, hasil);
                    } else {
                        printf("Error: Pembagian dengan nol tidak diperbolehkan.\n");
                    }
                    break;
            }
        } else if (pilihan == 5) {
            printf("Terimakasih, telah menggunakan kalkulator Muhammad Fadillah.\n");
        } else {
            printf("Input anda salah, silahkan coba lagi.\n");
        }
    } while (pilihan != 5);

    return 0;
}
